src/mpyl/build.py

```diff
@@ -239,17 +239,32 @@ def run_build(
 ):
     try:
         for stage, projects in accumulator.run_plan.items():
-            for proj in projects:
-                result = executor.execute(stage.name, proj, dry_run)
+            commands = [
+                ParallelObject(
+                    function=executor.execute,
+                    parameters={
+                        "stage": stage.name,
+                        "project": project,
+                        "dry_run": dry_run,
+                    },
+                )
+                for project in projects
+            ]
+            results: list[StepResult] = run_in_parallel(
```

Comment:
So if I understand correctly, we fail per _stage_. If one of the steps fails immediately, the others are allowed to complete?

Reply:
I think that strategy indeed makes the most sense. We could also terminate the other threads as soon as the first one fails, but we would need to introduce another "terminated" fail state. 
And although your build might run a bit longer than strictly necessary, you will get (probably) cacheable outcomes. And you also get potentially useful information to find out what's wrong from multiple failed build outputs.

So yes, failing on stage level makes most sense to me too.